### ISSUE TITLE
Sharing common-types between java projects - enhancement idea

### ISSUE THREAD

**Issue by bloat** [real-logic/simple-binary-encoding#752]
Similar to #581 I have a project A which defines certain common types and messages.

Then I have other projects (B,C etc.) which have their own schemas and they each xi:include the common-types.xml from project A.

This leads to multiple copies of the generated classes for the common items in the jars for the different projects.

Would it make sense to allow configuring SbeTool to suppress output for the common classes when processing the schemas for the projects B and C?

In that way they can still include common-types.xml and generate correct classes for their specific messages, but they will pick up the generated classes by depending on jars from Project A. They would not have their own copies.

I would propose adding a property for SbeTool containing a list of regex patterns for types/messages/etc where output should be suppressed.

How does this sound? I can look to implement it at least for Java and see if it works for me.

Thanks in advance

**Comment by tmontgomery** [comment]
It sounds interesting. We can take a look at a PR and see.

**Comment by bloat** [comment]
Just to let you know, I've implemented this and its working for Java for my use case.

Unfortunately my company's open source contribution rules mean that it will take some time for me to be able to raise a PR, but I will start the process if you think it will be useful.

Otherwise feel free to close this issue. 

For the record what I've done is pretty simple, I subclassed PackageOutputWriter and gave the new class a regex which it checks against the name in the createOutput method. If there is a match it returns a Writer.nullWriter. Otherwise it calls super.createOutput.
